chain/client/src/client.rs

```diff
@@ -926,7 +926,10 @@ impl Client {
         }
 
         if status.is_new_head() {
-            self.shards_mgr.update_largest_seen_height(block.header().height());
+            let height = block.header().height();
+            self.shards_mgr.update_largest_seen_height(height);
+            let prune_height = height - 2 * self.config.epoch_length;
```

Comment:
I didn't mean this literally. For one, you need to use at least `saturating_sub` to make sure we don't underflow. Also it is better to extract the constant out

Reply:
Per our offline discussion, I changed to last finalized block height in https://github.com/near/nearcore/pull/3710/commits/95cc9e71fb27f7bdbae794838c88abcb40515d8c